platform/pulp/platform/fields.py

```diff
@@ -0,0 +1,96 @@
+"""
+Custom django fields provided by pulp platform
+"""
+import json
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import six, ugettext_lazy as _
+
+
+def _parse_json(json_string):
+    """
+    Parse a json string
+
+    :param json_string: A json string to parse
+    :type json_string: str
+
+    :return: A python object built from the json string
+    :rtype: Any
+    """
+    try:
+        return json.loads(json_string)
+    except Exception:
+        raise ValidationError(_("Invalid JSON stored in database"))
+
+
+class JSONField(models.TextField):
+    """
+    A custom django field to serialize dicts into a text field
+
+    """
+    def from_db_value(self, value, expression, connection, context):
+        """
+        Converts a value as returned by the database to a Python object
+
+        :param value: The value to convert to Python
+        :type value: Any
+
+        :param expression:
+        :param connection:
+        :param context:
```

Comment:
Might be prudent to call super() first and then decode that value.

Something like:
```
value = super(JSONField, self).from_db_value(value, expression, connection, context)
if isinstance(value, six.string_types):
    return self.to_python(value)
else:
    return value
```